herbert_core/utilities/classes/@serializable/serializable.m

```diff
@@ -384,36 +384,59 @@
-            %               associated with
+            %            -- cellarray of positional parameter names,
+            %               coinciding with the names of the properties the
+            %               function is called to set
             % old_keyval_compat
             %            -- if set to true, keys in varargin may have form
             %               '-keyN' in addition to 'keyN'. Deprecation
             %                warning is issued for this kind of names.
             % varargin   -- cellarray of the constructor inputs, in the
-            %               form, described above
+            %               form, described above.
+            %
+            % End of positional parameters list is established by finding
+            % in varargin the element, belonging to the
+            % positinal_param_names_list  (first key)
+            %
+            % If the same property is defined using positional parameter
+            % and as key-value pair, the key-val parameter value takes
+            % priority.
+            %
+            %
             % EXAMPLE:
-            % if class have the properties {'a1'=1(numeric), 'a2'='blabla'(char),
-            % 'a3'=sqw() 'a4=[1,1,1] (numeric), and these properties are independent
-            % properties returned by saveableFields() function as list {'a1','a2','a3','a4'}
-            % Then the list of input parameters
-            % set_positional_and_key_val_arguments(1,'blabla',an_sqw_obj,'blabla','a4',[1,0,0])
-            % sets up the three first arguments as positional parameters, for properties
-            % a1,a2 and a3 and a4 is set as key-value pair. 'blabla' is returned in
-            % remains.
+            % if class have the properties {'a1'=1, 'a2'='blabla',
+            % 'a3'=sqw() 'a4=[1,1,1], 'a5'=something} and these properties
+            % are the independent properties defining the state of the
+            % object and provided in positional_param_names_list as:
+            % ppp = {'a1','a2','a3','a4','a5'}
+            % Then the call to the function with the list of input parameters:
+            % varargin = {1,'blabla',an_sqw_obj,'a4',[1,0,0],'blabla'}
+            % in the form:
+            %>> [obj,remains] = set_positional_and_key_val_arguments(obj,ppp,false,varargin{:});
+            %
+            % sets up the three first arguments as positional parameters,
+            % for properties a1,a2 and a3, a4 is set as key-value pair,
+            % 'blabla' is returned in remains and property a5 remains
+            % unset.
+            %
             %
             [obj,remains] = ...
                 set_positional_and_key_val_arguments_(obj,...
                 positinal_param_names_list,old_keyval_compat,varargin{:});
             %
-            % Code sample to insert into new object constructor to use generic constructor:
+            % Simple Code sample to insert into new object constructor 
```

Comment:
Should this be here or in the docstring?

Reply:
As a developer I do not use docstring as its easier to look into code directly. (and be sure comments are not outdated). 
And user would certainly not be bothered with this.